docs/frontend-system/utility-apis/05-migrating.md

```diff
@@ -0,0 +1,138 @@
+- Make sure that this extension is referenced by your migrated plugin
+
+## Prerequisites
+
+This guide assumes that you first [upgrade your repo](../../getting-started/keeping-backstage-updated.md) to the latest release of Backstage. This ensures that you do not have to fight several types of incompatibilities and updates at the same time.
+
+## Dependency changes
+
+In this article we will discuss some interfaces that you used to import from the `@backstage/core-plugin-api` package. Those are now generally moved over to `@backstage/frontend-plugin-api`, so you will want to update both your `package.json` and your source code. This applies both in your `-react` package and your main plugin package.
+
+First `package.json`. The following commands are examples - note that they refer to `plugins/example`, which you'll have to update to the actual folder name that your package to migrate is in.
+
+```bash title="from your repo root"
+yarn --cwd plugins/example remove @backstage/core-plugin-api ;
+yarn --cwd plugins/example add @backstage/frontend-plugin-api
+```
+
+Now in all of the code files in that package:
+
+```tsx title="in your source code"
+/* highlight-remove-next-line */
+import { createApiRef } from '@backstage/core-plugin-api';
+/* highlight-add-next-line */
+import { createApiRef } from '@backstage/frontend-plugin-api';
+```
+
+These can typically be search-and-replaced wholesale - the interfaces in the new package are mostly identical to the old one. The `createApiRef` is just an example, and the same replacement makes sense for all of the other symbols from the core package as well.
+
+## React package interface and ref changes
+
+Let's begin with [your `-react` package](../../architecture-decisions/adr011-plugin-package-structure.md). The act of exporting TypeScript interfaces and API refs have not changed from the old system. You can typically keep those as-is. For illustrative purposes, this is an example of an interface and its API ref:
+
+```tsx title="in @internal/plugin-example-react"
+import { createApiRef } from '@backstage/frontend-plugin-api';
+
+/**
+ * Performs some work.
+ * @oublic
+ */
+export interface WorkApi {
+  doWork(): Promise<void>;
+}
+
+/**
+ * The work interface for the Example plugin.
+ * @public
+ */
+export const workApiRef = createApiRef<WorkApi>({
+  id: 'plugin.example.work',
+});
+```
+
+In this example, the plugin ID already follows <!-- TODO: Link --> the common naming convention. If it doesn't, you may want to consider renaming that ID at this point. Don't worry, this won't hurt consumers in the old frontend system since the ID is mostly used for debugging purposes there. In the new system, it's much more important and appears in app-config files and similar.
+
+Note at the top of the file that it uses the updated import from `@backstage/frontend-plugin-api` that we migrated in the previous section, instead of the old `@backstage/core-plugin-api`.
+
+## Plugin package changes
```

Comment:
Feeling that this might be better left to the plugin migration guide instead, and that we perhaps don't need a migration section for Utility APIs at all. Just to avoid having to jump around too much when following the guide. Let's ship this though and perhaps move this over to the plugin migration guide when we have that.